utils/s3/client.cc

```diff
@@ -89,6 +90,18 @@ future<> ignore_reply(const http::reply& rep, input_stream<char>&& in_) {
 client::client(std::string host, endpoint_config_ptr cfg, semaphore& mem, global_factory gf, private_tag, std::unique_ptr<aws::retry_strategy> rs)
         : _host(std::move(host))
         , _cfg(std::move(cfg))
+        , _creds_invalidation_timer([this] {
+            s3l.info("Invalidating creds");
+            _credentials = {};
+        })
+        , _creds_update_timer([this] {
+            (void)[this]->future<> {
+                s3l.info("Update creds in the background");
+                _credentials = co_await _creds_provider_chain.get_aws_credentials();
+                _creds_invalidation_timer.rearm(_credentials.expires_at);
+                _creds_update_timer.rearm(_credentials.expires_at - 1h);
```

Comment:
Could be just `_creds_update_timer.arm()` (because it's no longer armed here), but otherwise looks OK